app/Models/User.php

```diff
@@ -193,6 +196,32 @@ public function markEmailAsVerified(): bool
         ])->save();
     }
 
+    /**
+     * Determine if the user can access the Filament admin panel.
+     * This function checks if the user has the 'admin' role, which
+     * is required to grant access to Filament. Users without this
+     * role will be denied access.
+     *
+     * @return bool True if the user has the 'admin' role, false otherwise.
+     */
+    public function canAccessPanel(Panel $panel): bool
+    {
+	return $this->hasRole('admin');
```

Comment:
hey, we should use permissions here instead, like 'view_librarium' and add this new permission to the author role in the role and permission seeder.